src/hooks/useTransactionType.ts

```diff
@@ -1,90 +1,107 @@
-          text: isSendTx ? (isTxQueued(tx.txStatus) ? 'Send' : 'Sent') : 'Received',
-        })
-        break
+      return {
+        icon: isSendTx ? '/images/outgoing.svg' : '/images/incoming.svg',
+        text: isSendTx ? (isTxQueued(tx.txStatus) ? 'Send' : 'Sent') : 'Received',
       }
-      case 'SettingsChange': {
-        // deleteGuard doesn't exist in Solidity
-        // It is decoded as 'setGuard' with a settingsInfo.type of 'DELETE_GUARD'
-        const isDeleteGuard = tx.txInfo.settingsInfo?.type === SettingsInfoType.DELETE_GUARD
-        setType({ icon: '/images/settings.svg', text: isDeleteGuard ? 'deleteGuard' : tx.txInfo.dataDecoded.method })
-        break
-      }
-      case 'Custom': {
-        if (isModuleExecutionInfo(tx.executionInfo)) {
-          const DEFAULT_MODULE_NAME = 'Module'
-
-          const { logoUri, name } = tx.txInfo.to
+    }
+    case TransactionInfoType.SETTINGS_CHANGE: {
+      // deleteGuard doesn't exist in Solidity
+      // It is decoded as 'setGuard' with a settingsInfo.type of 'DELETE_GUARD'
+      const isDeleteGuard = tx.txInfo.settingsInfo?.type === SettingsInfoType.DELETE_GUARD
 
-          setType({
-            icon: logoUri || '/images/settings.svg',
-            text: name || DEFAULT_MODULE_NAME,
-          })
-          break
+      return {
+        icon: '/images/settings.svg',
+        text: isDeleteGuard ? 'deleteGuard' : tx.txInfo.dataDecoded.method,
+      }
+    }
+    case TransactionInfoType.CUSTOM: {
+      if (isModuleExecutionInfo(tx.executionInfo)) {
+        return {
+          icon: toAddress?.logoUri || '/images/settings.svg',
+          text: toAddress?.name || DEFAULT_MODULE_NAME,
         }
+      }
 
-        if (isCancellationTxInfo(tx.txInfo)) {
-          setType({ icon: '/images/circle-cross-red.svg', text: 'On-chain rejection' })
-          break
+      if (isCancellationTxInfo(tx.txInfo)) {
+        return {
+          icon: '/images/circle-cross-red.svg',
+          text: 'On-chain rejection',
         }
+      }
 
-        if (tx.safeAppInfo) {
-          setType({ icon: tx.safeAppInfo.logoUri, text: tx.safeAppInfo.name })
-          break
+      if (tx.safeAppInfo) {
+        return {
+          icon: tx.safeAppInfo.logoUri,
+          text: tx.safeAppInfo.name,
         }
+      }
 
-        setType({
-          icon: toAddress?.logoUri || '/images/custom.svg',
-          text: toAddress?.name || 'Contract interaction',
-        })
-        break
+      return {
+        icon: toAddress?.logoUri || '/images/custom.svg',
+        text: toAddress?.name || 'Contract interaction',
+      }
+    }
+    default: {
+      return {
+        icon: '/images/custom.svg',
+        text: 'Contract interaction',
       }
     }
-  }, [tx, toAddress?.logoUri, toAddress?.name])
+  }
+}
 
-  return type
+export const useTransactionType = (tx: TransactionSummary): TxType => {
+  return useMemo(() => {
+    return getTxType(tx)
+  }, [tx])
```

Comment:
`useTransactionType` is only being used in one place. Should we also remove it and call `getTxType` directly?

Reply:
That's a good point! I don't think it harms leaving it as it is for now though.